ERCS/erc-7562.md

```diff
@@ -0,0 +1,393 @@
+
+* **[AREP-010]** each alt-mempool manages "opsSeen" and "opsIncluded", much like entities. The opsSeen is incremented after `UserOperation` initial validation, where it is considered part of this mempool.
+    The "opsIncluded" is incremented after this UserOperation is included on-chain (either by this bundler, or another)
+* **[AREP-020]** the alt-mempool becomes THROTTLED based on the [Reputation Calculation](#reputation-calculation)
+
+## Rationale
+
+All transactions initiated by EOAs have an implicit validation phase where balance, nonce, and signature are
+checked to be valid for the current state of the Ethereum blockchain.
+Once the transaction is checked to be valid by a node, only another transaction by the same EOA can modify the Ethereum
+state in a way that makes the first transaction invalid.
+
+With Account Abstraction, however, the validation can also include an arbitrary EVM code and rely on storage as well,
+which means that unrelated `UserOperations` or transactions may invalidate each other.
+
+If not addressed, this would make the job of maintaining a mempool of valid `UserOperations` and producing valid
+bundles computationally infeasible and susceptible to DoS attacks.
+
+This document describes a set of validation rules that if applied by a bundler before accepting a `UserOperation`
+into the mempool can prevent such attacks.
+
+### The high-level goal
+
+The purpose of this specification is to define a consensus between nodes (bundlers or block-builders) when processing incoming UserOperations from an external source.
+This external source for UserOperations is either an end-user node (via RPC) or another node in the p2p network.
+
+The protocol tries to detect "spam" - which are large bursts of UserOperations that cannot be included on-chain (and thus can't pay).
+The network is protected by throttling down requests from such spammer nodes.
+
+All nodes in the network must have the same definition of "spam": otherwise, if some nodes accept some type of UserOperations and propagate them while others consider them spam, those "forgiving" nodes will be considered "spammers" by the rest of the nodes, and the network effectively gets split.
+
+### The processing flow of a UserOperation
+
+- First, a UserOperation is received - either via RPC (submitted on behalf of a single application) or via the p2p protocol, from another node in the mempool.
+- The node performs validation on the UserOperation, and then adds it to its in-memory mempool, and submits it to its peers.
+- Lastly, when building a block, a node collects UserOperations from the mempool, performs a 2nd validation to make sure they are all still valid as a bundle and submits them into the next block.
```

Comment:
This subsection looks like it belongs in the Specification section. It isn't really explaining why you've made the choices you've made. Instead, this text is demonstrating an algorithm. That makes it more Specification content.